### PR TITLE
New interoperability delta onboarding (2665 - 2762)

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR includes the delta onboarding. Part of it is the new screen, and delta onboarding logic with debug menu entry

🌟 This PR consists of code, which already had a review.

Stories: 
[2762](https://jira.itc.sap.com/browse/EXPOSUREAPP-2762)
[2761](https://jira.itc.sap.com/browse/EXPOSUREAPP-2761)

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [ ] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		}
+		return version
+	}
+
+	var appBuildNumber: String {
+		guard let buildNumber = infoDictionary?[kCFBundleVersionKey as String] as? String else {
+			fatalError("Could not read CFBundleVersion from Bundle.")
+		}
+		return buildNumber
+	}
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift:34]: No project without this! 😃🎉

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift [hunk 2]
```diff
@@ -60,13 +60,9 @@ extension AppInformationViewController {
...
-		if let bundleVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"],
-			let bundleBuild = Bundle.main.infoDictionary?["CFBundleVersion"] {
-			versionLabel.text = "\(AppStrings.Home.appInformationVersion) \(bundleVersion) (\(bundleBuild))"
-		} else {
-			versionLabel.text = "\(AppStrings.Home.appInformationVersion) <unknown>"
-			logError(message: "Unknown version. Should not happen!")
-		}
+		let bundleVersion = Bundle.main.appVersion
+		let bundleBuild = Bundle.main.appBuildNumber
+		versionLabel.text = "\(AppStrings.Home.appInformationVersion) \(bundleVersion) (\(bundleBuild))"
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:65]: I would question the need for build numbers in the production version. Normally not worth a discussion, People seem to be crazy about the simplest things here. 
I would suggest to omit the build number in prod builds and keep it for the devs and testers available, e.g. compiler flag.

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/V15/DeltaOnboardingV15.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import UIKit
+
+class DeltaOnboardingV15: DeltaOnboarding {
+
+	let version = "1.5"
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/V15/DeltaOnboardingV15.swift:24]: Just curious, where is this version from? The overall app version or just a coincidence?

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+import Foundation
+import XCTest
+@testable import ENA
+
+class DeltaOnboardingViewControllerTests: XCTestCase {
+	
+	override func setUp() {
+		super.setUp()
+	}
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift:28]: Needed?

```suggestion
```

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+			DeltaOnboardingV15ViewController(
+				coder: coder,
+				supportedCountries: []
+			)
+		}
+	}
+	
+	func testCellsInSection0() {
+		let vc = createVC()
+		_ = vc.view
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift:50]:
```suggestion
		vc.loadViewIfNeeded()
``` 
is cleaner to read and understand. Following appearances also.

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift [hunk 6]
```diff
@@ -35,6 +35,7 @@ final class MockTestStore: Store, AppConfigCaching {
 	var exposureActivationConsentAcceptTimestamp: Int64?
 	var exposureActivationConsentAccept: Bool = false
 	var isOnboarded: Bool = false
+	var onboardingVersion: String = ""
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift:38]: We should `nil` the heck out of these empty strings!

**Review by whiskey — CHANGES_REQUESTED** [review]
That's a lot of onboarding versions! :) 

Small suggestions. Happy to discuss.

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/V15/DeltaOnboardingV15.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/V15/DeltaOnboardingV15.swift:24]: To make delta onboarding work, it has to align with the corresponding app version.

File: src/xcode/ENA/ENA/Resources/Storyboards/Onboarding.storyboard [hunk 7]
```diff
@@ -1,10 +1,12 @@
...
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="16097.3" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="Vq1-Uv-hV6">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="17156" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="Vq1-Uv-hV6">
     <device id="retina6_5" orientation="portrait" appearance="light"/>
     <dependencies>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="16087"/>
+        <deployment identifier="iOS"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="17125"/>
         <capability name="Named colors" minToolsVersion="9.0"/>
         <capability name="Safe area layout guides" minToolsVersion="9.0"/>
+        <capability name="System colors in document resources" minToolsVersion="11.0"/>
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Resources/Storyboards/Onboarding.storyboard:9]: 👍

File: src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift:34]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:65]: Ok I think this is something for the refactoring list. And it also needs to be aligned with whoever had interest to have this in the app.

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift (hunk 4 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift:28]: removed.

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift (hunk 5 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingViewControllerTests.swift:50]: changed

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift (hunk 6 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift:38]: Its not optional and has a default value in the real store. So no nillable.

**Review by SwissMountainFrog — APPROVED** [review]

**Review by janetback — CHANGES_REQUESTED** [review]
Hi Artur, can you please change the following sentence:
"DeltaOnboarding_ParticipatingCountriesList_Unavailable" = "Die teilnehmenden Länder können Sie jederzeit einsehen in den Details zur Risiko-Ermittlung.";
Please change it so the verb is at the end:
"Die teilnehmenden Länder können Sie jederzeit in den Details zur Risiko-Ermittlung einsehen.";
- (this was a mistake in the original) Thanks.

**Review by janetback — APPROVED** [review]
UA reviewed

File: src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift (hunk 6 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Services/__tests__/Mocks/MockTestStore.swift:38]: Yes, but the store is crap with this empty String…

**Review by whiskey — APPROVED** [review]

**Pull request merged**